src/main/scala/org/constellation/checkpoint/CheckpointMerkleService.scala

```diff
@@ -105,8 +105,7 @@ class CheckpointMerkleService[F[_]: Concurrent](
       merkleRoot,
       observationService,
       (o: Observation) => o,
-      (s: List[String]) =>
-        ??? // TODO: wkoszycki Resolving in this place indicates IllegallState thus should be removed totally
+      (s: List[String]) => LiftIO[F].liftIO(DataResolver.resolveBatchObservationsDefaults(s)(contextShift)(dao = dao))
```

Comment:
@kpudlik Have you seen the comment above :D ??

Reply:
Yes, and Marcin have seen it as well as he removed it, which means that it doesn't produce IllegalState anymore right? :D

Reply:
We’re resolving transactions in such a way and once I’ve added observations to the block, I’ve seen NotImplementedException in the logs, which means we’re anyway using this code.

Reply:
That would indicate something is wrong because:

merkle roots are created locally, lack of data means that merkle root was created but data wasn't store or was deleted. It happened for transactions previously because I have stored resolved checkpointblock without storing it's content e.g transactions

Reply:
It might be due to not storing observations on redownload:

```
  def setSnapshot(snapshotInfo: SnapshotInfo): F[Unit] =
    for {
      _ <- retainOldData()
      _ <- snapshot.set(snapshotInfo.snapshot)
      _ <- lastSnapshotHeight.set(snapshotInfo.lastSnapshotHeight)
      _ <- concurrentTipService.set(snapshotInfo.tips)
      _ <- acceptedCBSinceSnapshot.set(snapshotInfo.acceptedCBSinceSnapshot)
      _ <- transactionService.transactionChainService.applySnapshotInfo(snapshotInfo)
      _ <- snapshotInfo.addressCacheData.map { case (k, v) => addressService.putUnsafe(k, v) }.toList.sequence
      _ <- (snapshotInfo.snapshotCache ++ snapshotInfo.acceptedCBSinceSnapshotCache).toList.map { h =>
        LiftIO[F].liftIO(h.checkpointBlock.get.storeSOE()) >>
          checkpointService.put(h) >>
          dao.metrics.incrementMetricAsync(Metrics.checkpointAccepted) >>
          h.checkpointBlock.get.transactions.toList
            .map(
              tx =>
                transactionService
                  .accept(TransactionCacheData(tx))
                  .flatTap(_ => dao.metrics.incrementMetricAsync("transactionAccepted"))
            )
            .sequence
      }.sequence
      _ <- dao.metrics.updateMetricAsync[F](
        "acceptedCBCacheMatchesAcceptedSize",
        (snapshotInfo.acceptedCBSinceSnapshot.size == snapshotInfo.acceptedCBSinceSnapshotCache.size).toString
      )
      _ <- updateMetricsAfterSnapshot()
    } yield ()```

checkpointservice put will produce merkleroot however there will be not stored further

Reply:
Thanks @wkoszycki